examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c

```diff
@@ -655,14 +721,169 @@ static sl_status_t wfx_rsi_do_join(void)
                 wfx_retry_interval_handler(is_wifi_disconnection_event, wfx_rsi.join_retries);
                 if (is_wifi_disconnection_event || wfx_rsi.join_retries <= MAX_JOIN_RETRIES_COUNT)
                 {
-                    xEventGroupSetBits(wfx_rsi.events, WFX_EVT_STA_START_JOIN);
+                    event.eventType = WFX_EVT_STA_START_JOIN;
+                    WfxPostEvent(&event);
                 }
             }
         }
     }
     return status;
 }
 
+void HandleDHCPPolling()
+{
+    struct netif * sta_netif;
+    WfxEvent_t event;
+
+    sta_netif = wfx_get_netif(SL_WFX_STA_INTERFACE);
+#if (CHIP_DEVICE_CONFIG_ENABLE_IPV4)
+    uint8_t dhcp_state = dhcpclient_poll(sta_netif);
+    if (dhcp_state == DHCP_ADDRESS_ASSIGNED && !hasNotifiedIPV4)
+    {
+        wfx_dhcp_got_ipv4((uint32_t) sta_netif->ip_addr.u_addr.ip4.addr);
+        hasNotifiedIPV4 = true;
```

Comment:
Shouldn't the event DHCP_DONE called here as well.

event.eventType = WFX_EVT_STA_DHCP_DONE;